presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeGeneratorContext.java

```diff
@@ -72,6 +75,11 @@ public BytecodeNode generate(RowExpression expression)
         return bytecodeGenerator.generate(expression, scope);
     }
 
+    public BytecodeNode generate(RowExpression expression, Class lambdaInterface)
```

Comment:
Make the last argument `Optional<Class>`. Simplify callsite (to remote the `if`). Having the other `generate` method delegate to this one.